### PR TITLE
Support inline content for JDL

### PR DESCRIPTION
**Pull request by deepu105**
Makes it possible to invoke import-jdl with inline content

```
jhipster import-jdl --inline \
        "application {
            config {
                baseName jhapp,
                testFrameworks [protractor]
            }
        }"
```

or 

```
jhipster import-jdl --inline "application { config { baseName jhapp, testFrameworks [protractor] }}"
```

-   Please make sure the below checklist is followed for Pull Requests.

-   [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: cli/import-jdl.js [hunk 1]
```diff
@@ -248,23 +252,14 @@ const shouldTriggerInstall = (generator, index) =>
 const isAllCompleted = items => Object.values(items).every(it => it);
 
 class JDLProcessor {
-    constructor(jdlFiles, options) {
-        logger.debug(`JDLProcessor started with jdlFiles: ${jdlFiles} and options: ${toString(options)}`);
+    constructor(jdlFiles, jdlContent, options) {
+        logger.debug(`JDLProcessor started with content: ${jdlFiles} and options: ${toString(options)}`);
```
Comment (MathieuAA) [inline cli/import-jdl.js]: I don't think the log helps a lot in this case. Maybe telling that it started with files and/or content and options is a good idea. For instance: if I call the JDLProcessor with only files, the log tells me that the processor started with files and options (no content); or if only with a content that the processor started with only a content and options.

File: package.json [hunk 2]
```diff
@@ -59,7 +59,7 @@
         "glob": "7.1.3",
         "gulp-filter": "5.1.0",
         "insight": "0.10.1",
-        "jhipster-core": "4.3.0",
+        "jhipster-core": "4.4.0",
```
Comment (MathieuAA) [inline package.json]: 5.0.0 might be better here (same content)

File: cli/import-jdl.js (hunk 1 above)
Comment (deepu105) [reply cli/import-jdl.js]: Sure

File: package.json (hunk 2 above)
Comment (deepu105) [reply package.json]: What about the breaking changes? Don't we have to update anything in the generator?

Comment (MathieuAA) [reply package.json]: Nop! this is the good kind of breaking change, doesn't impact the generator.

Comment (deepu105) [reply package.json]: great, what about the custom annotation stuff, I saw a comment somewhere discussing aout changing the entity json files to have options inside an `options` key? was that done?

Comment (deepu105) [reply package.json]: Also for the `SEARCH_ENGINE` change doesn't it break existing JDL files? if it has `searchEngine` in it? given that we will update and release this in a minor JHipster version may be we need to add backward compatibility for JDL files that still uses `searchEngine`. WDYT?

Comment (MathieuAA) [reply package.json]: @deepu105 about the first question, I didn't see any comment so I can't relate, nor have I seen work about this `options` key.
About the second one: in the JDL, you still use the `search` keyword. Work has been done to harmonize what's written in a JDL file and the internal name of the option holding the value. The breaking change is about what's exposed by BinaryOptions.

Comment (MathieuAA) [reply package.json]: To be honest, the first point bothers me. @yelhouti do you have any input about this? I don't remember about such a key :(

Comment (murdos) [reply package.json]: @MathieuAA : the `options` key have been suggested by you in [this comment](https://github.com/jhipster/jhipster-core/issues/320#issuecomment-485381566), but this has not been done (yet?).

Comment (MathieuAA) [reply package.json]: @murdos ahah! yep. Now I remember. That was a suggestion for using custom annotations, but doesn't break existing options

Comment (deepu105) [reply package.json]: perfect

Comment (yelhouti) [reply package.json]: @MathieuAA @deepu105 , options field is only used for custom annotations for backward compatility reasons, once we switch to JDL centric approach, I could go through the code again to harmonize the logique.

Comment (MathieuAA) [reply package.json]: @yelhouti nice :)

**Pull request merged**